### ISSUE TITLE
Rate limit handling does not work for file uploads with streams

### ISSUE THREAD

**Issue by qeled** [hydrabolt/discord.js#268]
Currently the library does not handle rate limited file uploads with streams correctly and attempts to pass a destroyed stream into superagent:

```js
process.on('unhandledRejection', function(error, promise) {
	console.error("Unhandled rejection:");
	console.error(error && error.stack ? error.stack : error);
});

// send 10 messages to consume all tokens from message bucket
var channel = client.channels.get("id", "<channelid>");
var pending = [];
for (var i = 0; i < 10; i++) {
  var p = channel.sendMessage("#" + i);
  pending.push(p);
}

// send a file request that will get rate limited
Promise.all(pending).then(() => {
  channel.sendFile(fs.createReadStream("test.png"));
});

// request will be sent again, but the stream is not readable anymore

// after a while connection times out because the stream cannot be
// read more than 1 time:

// Unhandled rejection:
// Error: read ECONNRESET
//     at exports._errnoException (util.js:874:11)
//     at TLSWrap.onread (net.js:544:26)
```

**Comment by abalabahaha** [comment]
i think resolve the file only when making the request?

**Comment by hydrabolt** [comment]
Should be fixed in [9.0.2](https://github.com/hydrabolt/discord.js)
